lib/master_manipulator/service.rb

```diff
@@ -2,71 +2,226 @@
 
 module MasterManipulator
   module Service
-
     # Restart the puppet server and wait for it to come back up
     # ==== Attributes
     # *+host+ - the host that this should operate on
     # *+opts+ - an options hash - not required
     #   *+:wait_cycle+ - the number of cycles to attempt retry
     #   *+:is_pe?+ - Boolean : if the SUT is PE, defaults to true
+    #   *+:hup?+ - Boolean : if true, reload server; if false, restart server; defaults to false
     #
     # Raises a standard error if the wait is unsuccessful
     #
     # ==== Example
     # restart_puppet_server(master)
     # restart_puppet_server(master, {:wait_cycles => 20})
-    def restart_puppet_server(host, opts = {})
-
+    # restart_puppet_server(master, {:is_pe? => false, :hup? => true})
+    def restart_puppet_server(host, opts = {:is_pe? => true, :hup? => false})
       start_time = Time.now
 
-      opts[:is_pe?] ||= true
       opts[:is_pe?] ? service_name = 'pe-puppetserver' : service_name = 'puppetserver'
-
-      on(host, puppet("resource service #{service_name} ensure=stopped"))
-      on(host, puppet("resource service #{service_name} ensure=running"))
-      hostname = on(host, 'hostname').stdout.chomp
-      opts[:wait_cycles] ||= 10
-
-      pe_ver = pe_version(host)
-      three_eight_regex = /^3\.8/
-
-      # This logic is not ideal refactor in the future
-      # if its PE and not 3.8 use the status endpoint
-      # it its not PE or it is 3.8 use the simple curl call
-      if pe_ver && !pe_ver.match(three_eight_regex)
-        curl_call = "-k -X GET -H 'Content-Type: application/json' https://#{hostname}:8140/status/v1/services?level=debug"
+    
+      if opts[:hup?]
```

Comment:
Oh, I see. Well, it's a boolean (line 13, in the comment), not a string. Or at least that's what I intended. I tested this before I checked it in.
```
irb(main):001:0> hup = false
=> false
irb(main):002:0> if hup { puts "or not" }
irb(main):003:1>
```